### DISCUSSION TITLE
Integrate UpSnap with Rustdesk

### DISCUSSION THREAD

**Discussion by lexcorp** [seriousm4x/UpSnap#556] [Feature requests]
Please integrate UpSnap with Rustdesk, add a button on card device to remote desktop connect via rustdesk. And a section or form to configure id of rustdesk, ip for device.

![UpSnap_Rustdeskv2](https://github.com/seriousm4x/UpSnap/assets/5751288/02771107-16ea-4876-85d4-ed0903ac6d87)

![image](https://github.com/seriousm4x/UpSnap/assets/5751288/cdd25109-39d1-4eec-8954-4fd00bfc91bb)

**Comment by lexcorp** [comment]
@seriousm4x Maybe this help:


[https://github.com/glpi-project/glpi-agent/blob/develop/lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm](https://github.com/glpi-project/glpi-agent/blob/develop/lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm)

**Comment by lexcorp** [comment]
In GLPI there is a remote desktop module:

![image](https://github.com/user-attachments/assets/93c2ec01-74da-4165-a36e-ac34ea518efb)

On my windows client I have rustdesk installed
![image](https://github.com/user-attachments/assets/13fb766b-007b-4259-b45a-61507aa60e5f)

The client ID matches the ID registered in GLPI
![image](https://github.com/user-attachments/assets/5ef95f22-6ae0-4674-9123-61a53bc13511)

Checking a little I notice that the generated URL has the format: rustdesk://1897204519

`<a href="rustdesk://1897204519" target="_blank">1897204519</a>`

![image](https://github.com/user-attachments/assets/7cee35dc-c4a5-4d0e-986d-6b2ebf26339e)

When I click the link, It shows me the option to open the rustdesk client:

![image](https://github.com/user-attachments/assets/426f8b97-1843-48b3-a367-1c4de393129c)

Upon accepting, the Rustdesk client opens to start the connection:
![image](https://github.com/user-attachments/assets/d4088000-e005-4d1b-9f4c-67538fb6108d)

**Comment by lexcorp** [comment]
I think you need to add somewhere in the form, an option to store the rustdesk client ID, like GLPI:

![image](https://github.com/user-attachments/assets/b33e57c6-e15b-45c2-b63f-7c2ed6d73c19)

![image](https://github.com/user-attachments/assets/7804d434-0b41-4ea2-9f49-61e000fee54a)

And be able to show it on the computer card in upsnap
![image](https://github.com/user-attachments/assets/fc6ec566-4a8a-48b1-9b28-6071a515376c)

**Comment by seriousm4x** [comment]
I have changed the field type from `url` to `string` so you can set the link to anything now (`rustdesk://some-id` or `rdp://some-ip`).
This is for the device name as well as the port link.

Will be available in the next release.
